administrator/components/com_media/Controller/Plugin.php

```diff
@@ -0,0 +1,187 @@
+<?php
+/**
+ * @package     Joomla.Administrator
+ * @subpackage  com_media
+ *
+ * @copyright   Copyright (C) 2005 - 2017 Open Source Matters, Inc. All rights reserved.
+ * @license     GNU General Public License version 2 or later; see LICENSE.txt
+ */
+
+namespace Joomla\Component\Media\Administrator\Controller;
+
+defined('_JEXEC') or die;
+
+use Joomla\CMS\Controller\Controller;
+use Joomla\CMS\Plugin\PluginHelper;
+use Joomla\Component\Media\Administrator\Event\OAuthCallbackEvent;
+use League\Flysystem\Exception;
+
+/**
+ * Plugin Controller for OAuth2.0 callbacks
+ *
+ * This controller is used for handle OAuth2 Callbacks
+ *
+ * @since  __DEPLOY_VERSION__
+ */
+class Plugin extends Controller
+{
+	/**
+	 * Handles an OAuth Callback request for a specified plugin
+	 *
+	 * URLs containing [sitename]/administrator/index.php?option=com_media&task=plugin.oauthcallback
+	 *  &plugin=[plugin_name]&status=[csrf_token]&....
+	 *
+	 * Will be handled by this endpoint.
+	 * It will select the plugin specified by plugin_name and pass all the data received from the provider
+	 *
+	 * @return void
+	 *
+	 * @since  __DEPLOY_VERSION__
+	 */
+	public function oauthcallback()
+	{
+		// Get the input
+		$input = $this->input->request->getArray();
+
+		// Check if CSRF was set in status or reject
+		if (!isset($input['status']))
```

Comment:
So what happens when a cloud doesn't support the status flag? I guess it will not work.